### ISSUE TITLE
Add documentation about import for definePageMeta

### ISSUE THREAD

**Issue by sky-code** [nuxt/nuxt#21306]
### Describe the feature

Please add documentation on how to import `definePageMeta`

```
import { definePageMeta } from '#imports'
```

to [definePageMeta](https://nuxt.com/docs/api/utils/define-page-meta#definepagemeta) page.

Without this import I have 

```
ReferenceError: definePageMeta is not defined
```

and it's very hard to find in documentation.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment]
Would you provide a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

You should not need to import this.

**Comment by sky-code** [comment]
Everything works in stackblitz, I can't reproduce

**Comment by sky-code** [comment]
Console logs
[localhost-1685673317351.log](https://github.com/nuxt/nuxt/files/11631289/localhost-1685673317351.log)

Regular options component with next structure:

```
import ...

definePageMeta({
    auth: false,
    layout: 'anon',
})

export default defineNuxtComponent({
...
})
```

**Comment by curtgrimes** [comment]
Here is a minimal reproduction of the issue with the latest version of Nuxt (3.5.3): https://github.com/curtgrimes/nuxt-definePageMeta-reproduction

<img width="625" alt="CleanShot 2023-06-05 at 23 38 11@2x" src="https://github.com/nuxt/nuxt/assets/3038600/68eb727a-5909-43a5-81b6-a1d28b28d720">

I'm not very familiar with how Nuxt 3 auto imports things, but when I look at my .nuxt/imports.d.ts file in this repo, I see the following, which does not mention `definePageMeta()` -- should it?

```ts
export { isVue2, isVue3 } from 'vue-demi';
export { useAsyncData, useLazyAsyncData, useNuxtData, refreshNuxtData, clearNuxtData, defineNuxtComponent, useNuxtApp, defineNuxtPlugin, definePayloadPlugin, reloadNuxtApp, useRuntimeConfig, useState, useFetch, useLazyFetch, useCookie, useRequestHeaders, useRequestEvent, useRequestFetch, useRequestURL, setResponseStatus, setPageLayout, onNuxtReady, useRouter, useRoute, defineNuxtRouteMiddleware, navigateTo, abortNavigation, addRouteMiddleware, showError, clearError, isNuxtError, useError, createError, defineNuxtLink, useAppConfig, updateAppConfig, defineAppConfig, preloadComponents, preloadRouteComponents, prefetchComponents, loadPayload, preloadPayload, isPrerendered, definePayloadReducer, definePayloadReviver, onBeforeRouteLeave, onBeforeRouteUpdate } from '#app';
export { withCtx, withDirectives, withKeys, withMemo, withModifiers, withScopeId, onActivated, onBeforeMount, onBeforeUnmount, onBeforeUpdate, onDeactivated, onErrorCaptured, onMounted, onRenderTracked, onRenderTriggered, onServerPrefetch, onUnmounted, onUpdated, computed, customRef, isProxy, isReactive, isReadonly, isRef, markRaw, proxyRefs, reactive, readonly, ref, shallowReactive, shallowReadonly, shallowRef, toRaw, toRef, toRefs, triggerRef, unref, watch, watchEffect, watchPostEffect, watchSyncEffect, isShallow, effect, effectScope, getCurrentScope, onScopeDispose, defineComponent, defineAsyncComponent, resolveComponent, getCurrentInstance, h, inject, hasInjectionContext, nextTick, provide, defineModel, defineOptions, defineSlots, mergeModels, toValue, useModel, useAttrs, useCssModule, useCssVars, useSlots, useTransitionState } from 'vue';
export { injectHead, useHead, useSeoMeta, useHeadSafe, useServerHead, useServerSeoMeta, useServerHeadSafe } from '@unhead/vue';
```

The .nuxt/types/imports.d.ts file also does not have any mention of `definePageMeta()` but I am not sure if it should.

**Comment by danielroe** [comment]
definePageMeta does not exist unless you have pages (and it is only usable within pages). 

It is a compiler macro and its contents become part of the route meta for that page.

**Comment by curtgrimes** [comment]
Thank you for the clarification. A combination of that and using `<script setup lang="ts">` instead of just `<script>` in my component was the issue.

**Comment by awadhjy** [comment]
> Thank you for the clarification. A combination of that and using `<script setup lang="ts">` instead of just `<script>` in my component was the issue.

but it still an issue for type script

**Comment by josueayala27** [comment]
> definePageMeta does not exist unless you have pages (and it is only usable within pages).
> 
> It is a compiler macro and its contents become part of the route meta for that page.

this was a very valuable clarification, should it be in the documentation?

**Comment by danielroe** [comment]
PR very welcome 🙏

**Comment by shidcordero** [comment]
@danielroe I am also having issue with typescript
![image](https://github.com/nuxt/nuxt/assets/26376208/4ba25a19-602f-45ea-bdc9-b2ca61f24bc4)

If I cast 'blank' as LayoutKey, it still won't work and cannot find my layout.

blank.vue was as simple as this

```
<script lang="ts" setup>
import { ref, onMounted } from 'vue'

const loading = ref(true)
onMounted(() => {
  loading.value = false
})
</script>

<template>
  <div class="h-screen">
    <div v-if="loading" class="loader">
      <VpLoading
        style="width: 50px; height: 50px;"
      />
    </div>
    <div v-else class="min-h-screen flex flex-column">
      test
      <div class="flex flex-column md:flex-row flex-auto">
        <div class="m-2 border-2 border-dashed surface-border border-round surface-section flex-auto p-5">
          <slot />
        </div>
      </div>
    </div>
  </div>
</template>
```

**Comment by josueayala27** [comment]
> @danielroe I am also having issue with typescript
> 
> ![image](https://github.com/nuxt/nuxt/assets/26376208/4ba25a19-602f-45ea-bdc9-b2ca61f24bc4)
> 
> 
> 
> If I cast 'blank' as LayoutKey, it still won't work and cannot find my layout.
> 
> 
> 
> blank.vue was as simple as this
> 
> 
> 
> ```
> 
> <script lang="ts" setup>
> 
> import { ref, onMounted } from 'vue'
> 
> 
> 
> const loading = ref(true)
> 
> onMounted(() => {
> 
>   loading.value = false
> 
> })
> 
> </script>
> 
> 
> 
> <template>
> 
>   <div class="h-screen">
> 
>     <div v-if="loading" class="loader">
> 
>       <VpLoading
> 
>         style="width: 50px; height: 50px;"
> 
>       />
> 
>     </div>
> 
>     <div v-else class="min-h-screen flex flex-column">
> 
>       test
> 
>       <div class="flex flex-column md:flex-row flex-auto">
> 
>         <div class="m-2 border-2 border-dashed surface-border border-round surface-section flex-auto p-5">
> 
>           <slot />
> 
>         </div>
> 
>       </div>
> 
>     </div>
> 
>   </div>
> 
> </template>
> 
> ```

to make sure, are you adding blank inside layouts folder?

**Comment by angelhdzmultimedia** [comment]
> Please add documentation on how to import definePageMeta

I came here but for other reason. I disabled auto imports (I've seen many people driving away from auto imports in favor of explicit imports, and wanted to feel what's working with Nuxt without auto imports before taking a decision) and now I struggled finding from where I could import it. In `unplugin-vue-router`, we can import `definePage` from the package.

So I searched in `nuxt/app`, `nuxt/schema`, `nuxt/config`, `#imports` (but then realized because I disabled auto imports then `#imports` virtual is gone), ultimately found it in `~/../.nuxt/imports`. So I agree with OP that the docs need more love regarding types/macros importing. 

Also curious about what was the reasoning for naming it `definePageMeta` instead of `definePage` just like `unplugin-vue-router` which is what we use when we work in a pure Vue project? 

Greetings to the Nuxt team. 🔥💚👋
